### ISSUE TITLE
cannot see each other

### ISSUE THREAD

**Issue by Dr1rD** [livekit/components-js#875]
### Select which package(s) are affected

\@livekit/components-styles, \@livekit/components-react, examples

### Describe the bug

I deployed livekit-web based on the example of components-js. Users open the livekit-web address and join a room, but can't see each other. What went wrong, please?

livekit web code snippet:

![微信图片_20240530142347](https://github.com/livekit/components-js/assets/32828866/2ba4b010-0e8b-4176-a481-797ac96d7bb8)

livekit server :

![微信图片_20240530143113](https://github.com/livekit/components-js/assets/32828866/19ee1fec-f2a6-4831-826f-1f1a1c0038ae)

### Reproduction

as above

### System Info

```shell
liverkit server:
    version: livekit_1.6.1_linux_amd64
    deploy:  virtual machine
```

### Severity

annoyance

**Comment by lukasIO** [comment]
Please provide client logs for both participants. It sounds more like a token issue than a issue with components. 
you can also test your tokens are valid with https://meet.livekit.io/?tab=custom

**Comment by Dr1rD** [comment]
client logs

![微信图片_20240531140958](https://github.com/livekit/components-js/assets/32828866/538a1878-a4d8-44fa-8448-620ddbedcbeb)
![微信图片_20240531141009](https://github.com/livekit/components-js/assets/32828866/fc74b5dd-eec8-45eb-befa-66ad326e62b3)

server logs

![微信图片_20240531141042](https://github.com/livekit/components-js/assets/32828866/482bcd46-e437-491f-ba1b-90988fc41dc3)


![微信图片_20240531141030](https://github.com/livekit/components-js/assets/32828866/fb2d643b-d849-4a75-9058-c2a8c41a7625)
![微信图片_20240531141022](https://github.com/livekit/components-js/assets/32828866/d983f6ee-1a01-4b1f-966d-76ad57bcf044)

**Comment by lukasIO** [comment]
It's really hard to parse logs from screenshots. Please post them as text

**Comment by Dr1rD** [comment]
Here are logs. Thank you!

[FROM_PCserver.log](https://github.com/user-attachments/files/15514547/FROM_PCserver.log)
[FROM_PCuser_dongrui_console.log](https://github.com/user-attachments/files/15514554/FROM_PCuser_dongrui_console.log)
[FROM_PCuser_hefj_console.log](https://github.com/user-attachments/files/15514558/FROM_PCuser_hefj_console.log)

**Comment by lukasIO** [comment]
Thank you, have you tried  https://meet.livekit.io/?tab=custom with some of your tokens, as I suggested?

**Comment by Dr1rD** [comment]
I have tried meet, but still the same problem.
I can subscribe the demo video stream which published by livekit-cli, but unable to publish my  video stream.

**Comment by lukasIO** [comment]
Please post the code you use for generating your tokens. Make sure they have publishing grants.

**Comment by Dr1rD** [comment]
Here are logs.
[server-20240606.log](https://github.com/user-attachments/files/15614199/server-20240606.log)
[client-robot.log](https://github.com/user-attachments/files/15614212/client-robot.txt)
[dongrui-create-token.txt](https://github.com/user-attachments/files/15614216/dongrui-token.txt)
[client-dongrui.log](https://github.com/user-attachments/files/15614218/client-dongrui.log)

I deploy server on VM by command  "./livekit-server --dev --bind private-ip".

User join the room via meet.  
User can subscribe the demo video stream which published by livekit-cli, but unable to publish user's video stream.

**Comment by lukasIO** [comment]
you're missing `canPublish` grant in your publisher token. see https://docs.livekit.io/home/get-started/authentication/#Video-grant

if you want to have users watch it they need `canSubscribe` respectively.
